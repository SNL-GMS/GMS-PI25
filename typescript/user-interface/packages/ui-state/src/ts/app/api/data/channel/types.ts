import type { AsyncFetchHistory } from '../../../query';

export interface GetChannelsByNamesTimeRangeQueryArgs {
  channelNames: string[];
  startTime: number;
  endTime: number;
}

export type GetChannelsByNamesHistory = AsyncFetchHistory<GetChannelsByNamesTimeRangeQueryArgs>;
