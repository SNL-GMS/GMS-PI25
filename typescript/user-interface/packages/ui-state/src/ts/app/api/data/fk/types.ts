/**
 * Input type for Compute FK service call. This input
 * is compatible with the COI input i.e. start/end are strings
 */
import type { FkTypes } from '@gms/common-model';

import type { AsyncFetchHistory } from '../../../query';

/**
 * Defines the history record type for the getChannelSegmentsByChannel query
 */
export type ComputeFkSpectraHistory = AsyncFetchHistory<FkTypes.FkInputWithConfiguration>;
