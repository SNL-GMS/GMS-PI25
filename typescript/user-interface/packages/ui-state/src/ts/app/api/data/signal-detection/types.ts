import type { EventTypes, SignalDetectionTypes, WorkflowTypes } from '@gms/common-model';
import type { EntityReference } from '@gms/common-model/lib/faceted';
import type {
  SignalDetectionHypothesis,
  SignalDetectionHypothesisFaceted
} from '@gms/common-model/lib/signal-detection';

import type { AsyncFetchHistory } from '../../../query';

export interface SignalDetectionId {
  id: string;
}

export interface GetFilterDefinitionsForSignalDetectionsQueryArgs {
  stageId: {
    name: string;
  };
  signalDetectionsHypotheses: SignalDetectionHypothesisFaceted[];
}
export interface GetFilterDefinitionsForSignalDetectionHypothesesQueryArgs {
  stageId: {
    name: string;
  };
  // In the case of a locally created signal detection we will need to query with the fully populated sdh
  signalDetectionsHypotheses: SignalDetectionHypothesisFaceted[] | SignalDetectionHypothesis[];
  eventHypothesis?: EventTypes.EventHypothesis;
}

export type GetFilterDefinitionsForSignalDetectionsHistory = AsyncFetchHistory<
  GetFilterDefinitionsForSignalDetectionsQueryArgs
>;

export type GetFilterDefinitionsForSignalDetectionHypothesesHistory = AsyncFetchHistory<
  GetFilterDefinitionsForSignalDetectionHypothesesQueryArgs
>;
/**
 * The interface required to make a signal detection query by stations.
 */
export interface GetSignalDetectionsAndSegmentsByStationsAndTimeQueryArgs {
  /**
   * Entity references contain only the station name.
   */
  stations: EntityReference<'name'>[];

  /**
   * In seconds. This will get converted into a UTC time string by the AxiosTransformers.
   */
  startTime: number;

  /**
   * In seconds. This will get converted into a UTC time string by the AxiosTransformers.
   */
  endTime: number;

  /**
   * The stage for which to request signal detections by station.
   */
  stageId: WorkflowTypes.WorkflowDefinitionId;

  /**
   * A list of signal detections to exclude from the result (so a request
   * with the same times could return newer results, in the case of late-arriving
   * data for example).
   */
  excludedSignalDetections?: SignalDetectionTypes.SignalDetection[];
}

/**
 * The interface required to make a signal detection query by a single station.
 */
export type GetSignalDetectionsWithSegmentsByStationAndTimeQueryArgs = Omit<
  GetSignalDetectionsAndSegmentsByStationsAndTimeQueryArgs & {
    station: EntityReference<'name'>;
  },
  'stations'
>;

/**
 * Defines the history record type for the getSignalDetectionsAndSegmentsByStationAndTime query
 */
export type GetSignalDetectionsWithSegmentsByStationAndTimeHistory = AsyncFetchHistory<
  GetSignalDetectionsWithSegmentsByStationAndTimeQueryArgs
>;

/**
 * Arguments to modify existing Signal Detection
 */
export interface UpdateSignalDetectionArgs {
  readonly signalDetectionIds: string[];
  readonly isDeleted?: boolean;
  readonly arrivalTime?: {
    readonly value: number;
    readonly uncertainty: number;
  };
  readonly phase?: string;
  // amplitudeMeasurement: SignalDetectionTypes.AmplitudeMeasurementValue;
}
