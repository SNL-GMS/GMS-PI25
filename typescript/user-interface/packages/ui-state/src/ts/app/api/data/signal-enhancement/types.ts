import type { ChannelSegmentTypes, ChannelTypes, CommonTypes, EventTypes } from '@gms/common-model';
import type { BeamType } from '@gms/common-model/lib/beamforming-templates/types';
import type { VersionReference } from '@gms/common-model/lib/faceted';
import type { ProcessingMaskDefinitionsByPhaseByChannel } from '@gms/common-model/lib/processing-mask-definitions/types';
import type { StationGroup } from '@gms/common-model/lib/workflow/types';

import type { AsyncFetchHistory } from '../../../query';

// query args for GetDefaultFilterDefinitionByUsageForChannelSegments call
export interface GetDefaultFilterDefinitionByUsageForChannelSegmentsQueryArgs {
  interval: CommonTypes.TimeRange;
  channelSegments: ChannelSegmentTypes.ChannelSegmentFaceted[];
  eventHypothesis?: EventTypes.EventHypothesis;
}

export type GetDefaultFilterDefinitionByUsageForChannelSegmentsHistory = AsyncFetchHistory<
  GetDefaultFilterDefinitionByUsageForChannelSegmentsQueryArgs
>;

// query args for GetProcessingMaskDefinitions call
export interface GetProcessingMaskDefinitionsQueryArgs {
  stationGroup: StationGroup;
  channels: VersionReference<'name', ChannelTypes.Channel>[];
  processingOperations: ChannelSegmentTypes.ProcessingOperation[];
  phaseTypes: string[];
}

export interface GetProcessingMaskDefinitionsQueryResult {
  processingMaskDefinitionByPhaseByChannel: ProcessingMaskDefinitionsByPhaseByChannel[];
}

export type GetProcessingMaskDefinitionsHistory = AsyncFetchHistory<
  GetProcessingMaskDefinitionsQueryArgs
>;

// query args for GetBeamformingTemplates call
export interface GetBeamformingTemplatesQueryArgs {
  phases: string[];
  stations: VersionReference<'name'>[];
  beamType: BeamType;
}

export type GetBeamformingTemplatesHistory = AsyncFetchHistory<GetBeamformingTemplatesQueryArgs>;
