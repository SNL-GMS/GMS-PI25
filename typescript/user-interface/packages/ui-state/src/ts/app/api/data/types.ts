import type { BeamformingTemplateTypes, SignalDetectionTypes } from '@gms/common-model';
import type { ProcessingMaskDefinitionsByPhaseByChannel } from '@gms/common-model/lib/processing-mask-definitions/types';

import type {
  AssociationConflictRecord,
  ChannelRecord,
  EventsRecord,
  FilterDefinitionsForChannelSegmentsRecord,
  FilterDefinitionsForSignalDetectionsRecord,
  FilterDefinitionsRecord,
  FkChannelSegmentRecord,
  FkFrequencyThumbnailRecord,
  QcSegmentRecord,
  SignalDetectionsRecord,
  UIChannelSegmentRecord
} from '../../../types';
import type { GetChannelsByNamesHistory } from './channel/types';
import type {
  FindEventsByAssociatedSignalDetectionHypothesesHistory,
  GetEventsWithDetectionsAndSegmentsByTimeHistory
} from './event/types';
import type { ComputeFkSpectraHistory } from './fk/types';
import type {
  GetFilterDefinitionsForSignalDetectionHypothesesHistory,
  GetFilterDefinitionsForSignalDetectionsHistory,
  GetSignalDetectionsWithSegmentsByStationAndTimeHistory
} from './signal-detection/types';
import type {
  GetBeamformingTemplatesHistory,
  GetDefaultFilterDefinitionByUsageForChannelSegmentsHistory,
  GetProcessingMaskDefinitionsHistory
} from './signal-enhancement/types';
import type {
  FindQCSegmentsByChannelAndTimeRangeHistory,
  GetChannelSegmentsByChannelHistory
} from './waveform/types';

/**
 * Defines the Data slice state.
 */
export interface DataState {
  /** the channel segments - by unique channel name - populated by multiple queries */
  uiChannelSegments: UIChannelSegmentRecord;

  /** Fk channel segment records */
  fkChannelSegments: FkChannelSegmentRecord;

  /** Fk frequency thumbnails record */
  fkFrequencyThumbnails: FkFrequencyThumbnailRecord;

  /** the channels by effectiveTime populated by the getChannelsByNamesTimeRange query */
  channels: {
    raw: ChannelRecord;
    beamed: ChannelRecord;
    filtered: ChannelRecord;
  };
  /** the signal detections - by signal detection id - populated by multiple queries */
  signalDetections: SignalDetectionsRecord;
  /** the events - by time- populated by multiple queries */
  events: EventsRecord;
  /** the association conflicts for events in conflict */
  associationConflict: AssociationConflictRecord;
  /** designed filter definitions */
  filterDefinitions: FilterDefinitionsRecord;
  /** filter definitions for signal detections */
  filterDefinitionsForSignalDetections: FilterDefinitionsForSignalDetectionsRecord;
  /** filter definitions for signal detections for the open event */
  filterDefinitionsForSignalDetectionHypothesesEventOpen: FilterDefinitionsForSignalDetectionsRecord;
  /** filter definitions for signal detections for the open event */
  filterDefinitionsForSignalDetectionHypotheses: FilterDefinitionsForSignalDetectionsRecord;
  /** missing SignalDetectionsHypotheses id's for filter definitions for signal detections for the open event */
  missingSignalDetectionsHypothesesForFilterDefinitions: SignalDetectionTypes.SignalDetectionHypothesisFaceted[];
  /** filter definitions for channel segments */
  defaultFilterDefinitionByUsageForChannelSegments: FilterDefinitionsForChannelSegmentsRecord;
  /** filter definitions for channel segments for open event */
  defaultFilterDefinitionByUsageForChannelSegmentsEventOpen: FilterDefinitionsForChannelSegmentsRecord;
  /** qc segments by unique channel name */
  qcSegments: QcSegmentRecord;
  /** Processing Mask definitions */
  processingMaskDefinitions: Record<string, ProcessingMaskDefinitionsByPhaseByChannel[]>;
  /** Beamforming templates */
  beamformingTemplates: BeamformingTemplateTypes.BeamformingTemplatesByBeamTypeByStationByPhase;
  /** query history */
  queries: {
    /** the history record of the computeFkSpectra call */
    computeFkSpectra: ComputeFkSpectraHistory;
    /** the history record of the getChannelSegmentsByChannel query */
    getChannelSegmentsByChannel: GetChannelSegmentsByChannelHistory;
    /** the history record of the findQCSegmentsByChannelAndTimeRange query */
    findQCSegmentsByChannelAndTimeRange: FindQCSegmentsByChannelAndTimeRangeHistory;
    /** the history record of the getSignalDetectionWithSegmentsByStationAndTime query */
    getSignalDetectionWithSegmentsByStationAndTime: GetSignalDetectionsWithSegmentsByStationAndTimeHistory;
    /** the history record of the getEventsWithDetectionsAndSegmentsByTime query */
    getEventsWithDetectionsAndSegmentsByTime: GetEventsWithDetectionsAndSegmentsByTimeHistory;
    /** the history record of the findEventsByAssociatedSignalDetectionHypotheses query */
    findEventsByAssociatedSignalDetectionHypotheses: FindEventsByAssociatedSignalDetectionHypothesesHistory;
    /** the history record of the getChannelsByName query */
    getChannelsByNamesTimeRange: GetChannelsByNamesHistory;
    /** the history record of the getFilterDefinitionsByUsage query */
    getFilterDefinitionsForSignalDetections: GetFilterDefinitionsForSignalDetectionsHistory;
    /** the history record of the getFilterDefinitionsByUsage query */
    getFilterDefinitionsForSignalDetectionHypotheses: GetFilterDefinitionsForSignalDetectionHypothesesHistory;
    /** the history record of the getDefaultFilterDefinitionByUsageForChannelSegments query */
    getDefaultFilterDefinitionByUsageForChannelSegments: GetDefaultFilterDefinitionByUsageForChannelSegmentsHistory;
    /** the history record of the getProcessingMaskDefinitions query */
    getProcessingMaskDefinitions: GetProcessingMaskDefinitionsHistory;
    /** the history record of the getBeamformingTemplates query */
    getBeamformingTemplates: GetBeamformingTemplatesHistory;
  };
}

export type DataStateKeys = keyof DataState;
