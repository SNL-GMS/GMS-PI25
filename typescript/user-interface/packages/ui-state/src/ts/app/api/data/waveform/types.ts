import type { VersionReference } from '@gms/common-model/lib/faceted';

import type { AsyncFetchHistory } from '../../../query/types';

/**
 * The interface required by the waveform manager service to make a channel segment query by channels.
 */
export interface GetChannelSegmentsByChannelsQueryArgs {
  /**
   * In seconds. This will get converted into a UTC time string by the AxiosTransformers.
   */
  startTime: number;
  /**
   * In seconds. This will get converted into a UTC time string by the AxiosTransformers.
   */
  endTime: number;
  /**
   * The `channel-timerange` endpoint expects version references of the channels.
   */
  channels: VersionReference<'name'>[];
}

/**
 * The interface required to make a channel segment query by single channel.
 */
export type GetChannelSegmentsByChannelQueryArgs = Omit<
  GetChannelSegmentsByChannelsQueryArgs & { channel: VersionReference<'name'> },
  'channels'
>;

/**
 * Defines the history record type for the getChannelSegmentsByChannel query
 */
export type GetChannelSegmentsByChannelHistory = AsyncFetchHistory<
  GetChannelSegmentsByChannelQueryArgs
>;

/**
 * The interface required to make a qc segment query
 */
export interface FindQCSegmentsByChannelAndTimeRangeQueryArgs {
  startTime: number;
  endTime: number;
  channels: VersionReference<'name'>[];
}

/**
 * Defines the history record type for the FindQCSegmentsByChannelAndTimeRange query
 */
export type FindQCSegmentsByChannelAndTimeRangeHistory = AsyncFetchHistory<
  FindQCSegmentsByChannelAndTimeRangeQueryArgs
>;
