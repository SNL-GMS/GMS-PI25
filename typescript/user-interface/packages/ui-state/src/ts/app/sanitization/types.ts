export interface DevToolEnvConfig {
  enabled: boolean;
  sanitizationMessage?: string;
  envVar: string;
}
