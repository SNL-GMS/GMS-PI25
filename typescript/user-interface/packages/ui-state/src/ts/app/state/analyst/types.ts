import type { WaveformTypes } from '@gms/common-model';
import { createEnumTypeGuard } from '@gms/common-util';
import Immutable from 'immutable';

export enum AnalystKeyAction {
  ESCAPE = 'Escape',
  SAVE_OPEN_EVENT = 'Save Open Event',
  SAVE_ALL_EVENTS = 'Save All Events in Interval'
}

export const AnalystKeyActions: Immutable.Map<string, AnalystKeyAction> = Immutable.Map([
  ['Control+KeyS', AnalystKeyAction.SAVE_OPEN_EVENT],
  ['Control+Shift+KeyS', AnalystKeyAction.SAVE_ALL_EVENTS],
  ['Escape', AnalystKeyAction.ESCAPE]
]);

/**
 * The display mode options for the waveform display.
 */
export enum WaveformDisplayMode {
  DEFAULT = 'Default',
  MEASUREMENT = 'Measurement'
}
/**
 * Available waveform align types.
 */
export enum AlignWaveformsOn {
  TIME = 'Time',
  PREDICTED_PHASE = 'Predicted',
  OBSERVED_PHASE = 'Observed'
}

/**
 * Available waveform sort types.
 */
export enum WaveformSortType {
  distance = 'Distance',
  stationNameAZ = 'Station: A-Z',
  stationNameZA = 'Station: Z-A'
}

export type DisplayedMagnitudeTypes = Record<string, boolean>;

export const isAnalystKeyAction = createEnumTypeGuard(AnalystKeyAction);

/**
 * Measurement mode state.
 */
export interface MeasurementMode {
  /** The display mode */
  mode: WaveformDisplayMode;

  /**
   * Measurement entries that are manually added or hidden by the user.
   * The key is the signal detection id
   */
  entries: Record<string, boolean>;
}

/**
 * A list such that at each index, it indicates whether the filter at the
 * same index is within the hotkey cycle.
 */
export type HotkeyCycleList = boolean[];

/**
 * The location solution state.
 * Includes:
 *   * The selected location solution set and solution
 *   * The selected preferred location solution set and solution
 */
export interface LocationSolutionState {
  selectedLocationSolutionId: string;
  selectedLocationSolutionSetId: string;
  selectedPreferredLocationSolutionId: string;
  selectedPreferredLocationSolutionSetId: string;
}

export interface RequestStatus {
  url: string;
  isComplete?: boolean;
  error?: any;
}

/**
 * Type used for determining the analyst action on an event
 */
export type EventActionTypes = 'duplicate' | 'reject' | 'delete' | 'open' | 'close' | 'details';

/**
 * Type used for determining the analyst action on a signal detection
 */
export type SignalDetectionActionTypes =
  | 'reject associate'
  | 'delete'
  | 'details'
  | 'associate'
  | 'unassociate'
  | 'phase'
  | 'default phase'
  | 'current phase'
  | 'fk';

/**
 * union of action types
 */
export type ActionTypes = EventActionTypes | SignalDetectionActionTypes | null;

export interface ActionTarget {
  actionType: ActionTypes; // the type of action the analyst will take on an event or sd
  eventIds: string[]; // The ids of the events that are the target of a user's action
  signalDetectionIds: string[];
  previousActionTargets: string[]; // ids of the previous action targets, used for nested context menus to keep track of ids
}

export interface AnalystState {
  channelFilters: Record<string, WaveformTypes.WaveformFilter>;
  defaultSignalDetectionPhase: string;
  effectiveNowTime: number;
  hotkeyCycleOverrides: Record<string, Record<number, boolean>>;
  location: LocationSolutionState;
  measurementMode: MeasurementMode;
  requestTracker: {
    /* record keyed on ids */
    requests: Record<string, RequestStatus>;
    initiatedRequests: number;
    completedRequests: number;
    lastRequestId: string;
  };
  openEventId: string;
  actionTargets: ActionTarget;
  openLayoutName: string;
  currentPhase: string;
  sdIdsToShowFk: string[];
  selectedEventIds: string[];
  selectedFilterIndex: number | null;
  selectedFilterList: string;
  selectedSdIds: string[];
  selectedSortType: WaveformSortType;

  eventListOpenEventTriggered: boolean;
  mapOpenEventTriggered: boolean;
  alignWaveformsOn: AlignWaveformsOn;
  phaseToAlignOn: string;
  phaseSelectorPhaseList: string;
  phaseSelectorFavorites: Record<string, string[]>;
}
