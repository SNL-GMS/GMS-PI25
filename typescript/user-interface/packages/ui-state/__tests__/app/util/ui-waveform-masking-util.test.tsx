import {
  PD01Channel,
  processingMaskDefinition,
  processingMaskDefinitionsByPhaseByChannel1,
  qcSegment,
  qcSegment2Version,
  qcSegment3Version2,
  qcSegment4Version,
  qcSegment5Version,
  qcSegment6Version,
  qcSegmentVersion
} from '@gms/common-model/__tests__/__data__';
import { ProcessingOperation } from '@gms/common-model/lib/channel-segment/types';
import { renderHook } from '@testing-library/react-hooks';
import React from 'react';
import { Provider } from 'react-redux';
import type { AnyAction } from 'redux';
import type { MockStoreCreator } from 'redux-mock-store';
import createMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import type { AppState } from '../../../src/ts/app/store';
import {
  createProcessingMasksFromQCSegmentVersions,
  useCreateProcessingMasks
} from '../../../src/ts/app/util/ui-waveform-masking-util';
import { unfilteredSamplesUiChannelSegment } from '../../__data__';
import { appState } from '../../test-util';

const MOCK_UUID = 123456789;

jest.mock('@gms/common-util', () => {
  const actual = jest.requireActual('@gms/common-util');
  return {
    ...actual,
    epochSecondsNow: () => 100,
    uuid4: () => MOCK_UUID
  };
});
const MOCK_TIME = 1606818240000;
global.Date.now = jest.fn(() => MOCK_TIME);

describe('ui waveform masking util', () => {
  describe('createProcessingMasksFromQCSegmentVersions', () => {
    const segmentVersions = [
      qcSegmentVersion,
      qcSegment2Version,
      qcSegment3Version2,
      qcSegment4Version,
      qcSegment5Version
    ];

    it('is defined', () => {
      expect(createProcessingMasksFromQCSegmentVersions).toBeDefined();
    });
    it('filters out segments that do not match the mask definition', () => {
      const processingMasks = createProcessingMasksFromQCSegmentVersions(
        segmentVersions,
        processingMaskDefinition
      );
      expect(processingMasks[0].maskedQcSegmentVersions).toEqual([
        qcSegment3Version2,
        qcSegment5Version
      ]);
    });
    it('groups segments based on the threshold and sets teh start and end time to the min and max', () => {
      const processingMasks = createProcessingMasksFromQCSegmentVersions(
        segmentVersions.concat(qcSegment6Version),
        processingMaskDefinition
      );

      expect(processingMasks[0].maskedQcSegmentVersions).toEqual([
        qcSegment3Version2,
        qcSegment5Version
      ]);
      expect(processingMasks[0].startTime).toEqual(qcSegment3Version2.startTime);
      expect(processingMasks[0].endTime).toEqual(qcSegment5Version.endTime);
      expect(processingMasks[1].maskedQcSegmentVersions).toEqual([qcSegment6Version]);
      expect(processingMasks[1].startTime).toEqual(qcSegment6Version.startTime);
      expect(processingMasks[1].endTime).toEqual(qcSegment6Version.endTime);
    });
    it('generates a new UUID for the processing mask', () => {
      const processingMasks = createProcessingMasksFromQCSegmentVersions(
        segmentVersions.concat(qcSegment6Version),
        processingMaskDefinition
      );

      expect(processingMasks[0].id).toEqual(MOCK_UUID);
    });
    it('sets the effective at to now', () => {
      const processingMasks = createProcessingMasksFromQCSegmentVersions(
        segmentVersions.concat(qcSegment6Version),
        processingMaskDefinition
      );

      expect(processingMasks[0].effectiveAt).toEqual(Date.now());
    });

    it('sets the processing operation to that of the mask definition', () => {
      const processingMasks = createProcessingMasksFromQCSegmentVersions(
        segmentVersions.concat(qcSegment6Version),
        processingMaskDefinition
      );

      expect(processingMasks[0].processingOperation).toEqual(
        processingMaskDefinition.processingOperation
      );
    });
  });

  describe('useCreateProcessingMasks', () => {
    it('creates processing masks', async () => {
      const mockStoreCreator: MockStoreCreator<AppState, AnyAction> = createMockStore([thunk]);

      const store = mockStoreCreator({
        ...appState,
        data: {
          ...appState.data,
          channels: { ...appState.data.channels, raw: { [PD01Channel.name]: PD01Channel } },
          qcSegments: { [PD01Channel.name]: { [qcSegment.id]: qcSegment } },
          processingMaskDefinitions: {
            [PD01Channel.name]: [processingMaskDefinitionsByPhaseByChannel1]
          }
        }
      });

      const { result } = renderHook(() => useCreateProcessingMasks(), {
        wrapper: (props: React.PropsWithChildren<unknown>) => (
          <Provider store={store}>{props.children}</Provider>
        )
      });
      const returnValue = await result.current(
        unfilteredSamplesUiChannelSegment,
        ProcessingOperation.EVENT_BEAM,
        'P'
      );
      expect(returnValue).toMatchSnapshot();
    });
  });
});
