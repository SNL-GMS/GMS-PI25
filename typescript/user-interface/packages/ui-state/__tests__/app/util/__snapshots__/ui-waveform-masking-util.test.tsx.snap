// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui waveform masking util useCreateProcessingMasks creates processing masks 1`] = `
{
  "channel": {
    "canonicalName": "PDAR.PD01.SHZ/masked/0501421b22b559405b02a551b38508334a9c51af6ea3c20b524f93a7501eb9f4",
    "channelBandType": "SHORT_PERIOD",
    "channelDataType": "SEISMIC",
    "channelInstrumentType": "HIGH_GAIN_SEISMOMETER",
    "channelOrientationCode": "Z",
    "channelOrientationType": "VERTICAL",
    "configuredInputs": [
      {
        "effectiveAt": 1636503404,
        "name": "PDAR.PD01.SHZ",
      },
    ],
    "description": "Raw Channel created from ReferenceChannel 2fabc2d3-858b-3e85-9f47-e2ee72060f0b with version d767395c-850e-36f8-a6f2-a1c7398440e4 Masked samples removed.",
    "effectiveAt": 1636503404,
    "effectiveForRequestTime": 1636503405,
    "effectiveUntil": 1660701599.984,
    "location": {
      "depthKm": 0.0381,
      "elevationKm": 2.192,
      "latitudeDegrees": 42.7765,
      "longitudeDegrees": -109.58314,
    },
    "name": "PDAR.PD01.SHZ/masked/0501421b22b559405b02a551b38508334a9c51af6ea3c20b524f93a7501eb9f4",
    "nominalSampleRateHz": 20,
    "orientationAngles": {
      "horizontalAngleDeg": -1,
      "verticalAngleDeg": 0,
    },
    "processingDefinition": {
      "appliedQcSegmentCategoryAndTypes": {
        "category": "ANALYST_DEFINED",
        "type": "AGGREGATE",
      },
      "maskedSegmentMergeThreshold": 50,
      "processingOperation": "EVENT_BEAM",
    },
    "processingMetadata": {
      "CHANNEL_GROUP": "PD01",
    },
    "response": undefined,
    "station": {
      "name": "PDAR",
    },
    "units": "NANOMETERS_PER_COUNT",
  },
  "processingMasks": [],
}
`;
